packages/node_modules/@webex/internal-plugin-devices/test/integration/spec/webex.js

```diff
@@ -0,0 +1,42 @@
+import '@webex/internal-plugin-devices';
+
+import {assert} from '@webex/test-helper-chai';
+import sinon from 'sinon';
+import WebexCore from '@webex/webex-core';
+import testUsers from '@webex/test-helper-test-users';
+
+describe('plugin-devices', () => {
+  describe('Webex', () => {
+    let devices;
+    let user;
+    let webex;
+
+    before('create users', () => testUsers.create({count: 1})
+      .then(([createdUser]) => {
+        user = createdUser;
+      }));
+
+    beforeEach('create webex instance', () => {
+      webex = new WebexCore({
+        credentials: user.token
+      });
+
+      devices = webex.internal.devices;
+
+      return devices.register();
+    });
+
+    describe('#onBeforeLogout()', () => {
+      beforeEach('setup spy', () => {
+        sinon.spy(devices, 'unregister');
+      });
+
+      it('unregisters the device', () =>
+        webex.logout({noRedirect: true})
```

Comment:
Are the limitations behind mock webex the reason why this couldn't be a unit test?

Reply:
This replicates the way that `internal-plugin-wdm` does this test. It makes sense for it to be an integration test as it also interfaces with a plugin, `webex-core`, which is outside of its plugin directory/repository.